### DISCUSSION TITLE
SIDDUMP

### DISCUSSION THREAD

**Discussion by PM250B** [bytespiller/sidplaywx#4] [Ideas]
As an export feature. SIDDUMP could be added?

**Comment by bytespiller** [comment]
I'm thinking, rather than integrating it directly, I might add support for the user-defined custom commands that would appear in the playlist's context menu  (e.g., a "Custom" submenu at the bottom).

Users could then define any number of custom commands with its label, path to the external executable, and arguments.

Following built-in dynamic arguments (placeholders) for sending to the external tools would be available (at least):
- `{filename}` (just the song file name without extension)
- `{fileext}` (file extension, without dot)
- `{filepath}` (full path including the filename with extension, edge case: can be within the zip file too, e.g., c:\test.zip\song.sid)
- `{filedir}` (just a directory where the song file resides, edge case: if within a zip file then it'd be e.g., c:\test.zip)
- `{songname}` (song name as defined in its metadata)
- `{subsongZ}` (subsong integer, starts from 0)
- `{subsong}` (subsong integer, starts from 1)
- `{duration}` (song duration in seconds, either from the songlengths DB or fallback)
- `{durationMs}` (like `{duration}` but in milliseconds)
- `{author}`
- `{copyright}`

Parameters example for a custom SIDDUMP command:
`c:\path\siddump.exe {filepath} -a{subsongZ} -t{duration} > c:\path\dump_{filename}_sub_{subsong}.txt`

Would something like this cover this scenario?

**Reply by PM250B** [reply]
I think it's a good idea.
